test/connection-test.ts

```diff
@@ -590,6 +590,26 @@ describe('Connection', function() {
     }
   )
 
+  it('should clean up websocket connection if error after websocket is opened', async function() {
+    await this.api.disconnect();
+    // fail on connection
+    this.api.connection._subscribeToLedger = async () => {
+      throw new Error('error on _subscribeToLedger')
+    }
+    try {
+      await this.api.connect();
+    } catch (err) {
+      // _ws.close event listener should have cleaned up the socket when disconnect _ws.close is run on connection error
+      // do not fail on connection anymore
+      this.api.connection._subscribeToLedger = async () => {}
+      await new Promise((resolve, reject) => {
+        setTimeout(async () => {
```

Comment:
Why the setTimeout? Can you remove it?

Reply:
This connect() needs some time for the disconnect to run _ws.close() and clean up the connection or we will still get `RippleError: Websocket connection never cleaned up` in the CI tests. In a real world scenario, if that happened it would be ok, as the connection would eventually get cleaned up, and the error would technically be correct if it was not.

Reply:
Actually, figured it out, updated, thx.